### DISCUSSION TITLE
how can i master kali linux

### DISCUSSION THREAD

**Discussion by mullastz** [community/community#61138] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

which procedure can guide me to master kali linux effectively in every aspect

**Comment by enginestein** [comment]
The first step is to install Kali Linux on your machine. Next, learn about bash scripting and terminal of kali linux. After all this it won't be hard for you to learn about penetration testing and hacking tools available on kali linux. I have a course of bash scripting if you want I can provide the material or you can search for bash scripting tutorials on youtube, there are many. Do not try to be quick and fast. install kali, learn bash and then learn hacking, eventually you will learn everything about kali and other linux distro too.

**Reply by mullastz** [reply]
thank you for your advice enginestein Aryan i have allready install kali linux in my machine and i have allready covered basic linux command but i dont know how to use any kind of tools in kali linux, set kali linux lab for more practise,to solve kali linux troublenshooting to write scripting and many aspect so i was asking for your help if you can help to direct me on how i can learn and where i can find material to study by starting to cover one aspect to othere

**Reply by enginestein** [reply]
As you just mentioned you want to master kali efficiently from every aspect, that will not be accomplished with basic knowledge. You have covered basic linux commands it's good that you have started but do not think about tools until you master them. Just focus on one thing, other routes will be created automatically, focus on bash and learning linux commands (bash is linux command line language). Apart from this many courses are available on Udemy if you are eager to learn bash, Udemy is not very expensive as well. If you need any further help, you can tell me.

**Reply by mullastz** [reply]
i receive your email and now some how i get light thank for you for that god bless you,you say that you have material about bash scripting if you don't mind can you assist me that material please

**Reply by enginestein** [reply]
Sure, I will provide you the video URL links but maybe not here because this is community and open to everyone, you can mail me at aryapraneil@gmail.com

**Reply by mullastz** [reply]
ok

**Comment by ghostinhershell** [comment]
Hey there!  👋

Thanks for posting in the GitHub Community, @mullastz! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. Good luck!
